VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js

```diff
@@ -105,8 +129,45 @@ angular.module('virtoCommerce.catalogModule')
 				canExecuteMethod: function () {
 					return true;
 				}
-			},
+            },
+            {
+                name: "Add filter", icon: 'fa fa-filter',
+                executeMethod: function () {
+                    var newBlade = {
+                        id: "propertySelector",
+                        entityType: "product",
+                        properties: blade.currentEntities,
+                        controller: 'virtoCommerce.catalogModule.propertySelectorController',
+                        template: 'Modules/$(VirtoCommerce.Catalog)/Scripts/blades/property-selector.tpl.html',
+                        onSelected: function (includedProperties) {
+                            blade.filtered = true;
+
 
+                            _.each(blade.currentEntities,
+                                function (property) {
+                                    var foundProperty = _.find(includedProperties,
+                                        function (selectedProperty) {
+                                            return selectedProperty.id === property.id;
+                                        });
+                                    property.isSelected = foundProperty !== undefined ? true : false;
+                                });
+                        }
+                    };
+                    bladeNavigationService.showBlade(newBlade, blade);
+                },
+                canExecuteMethod: function () {
+                    return true;
+                }
+            },
+            {
+                name: "clear filter", icon: 'fa fa-undo',
```

Comment:
Use localization, and I think better to call Reset filter, as it is not clear all fields, but resets it to initial state with all fields selected.